python/cudf/cudf/_lib/csv.pyx

```diff
@@ -533,8 +533,14 @@ def write_csv(
         .build()
     )
 
-    with nogil:
-        cpp_write_csv(options)
+    try:
+        with nogil:
+            cpp_write_csv(options)
+    except OverflowError:
+        raise OverflowError(
+            f"Writing CSV file with chunksize={rows_per_chunk} failed. "
+            "Consider providing a smaller chunksize argument."
+        )
```

Comment:
JSON writer also could use similar code as above.

Reply:
can you update json.pyx as well?

Reply:
Done.